ledger-core/virtual/handlers/handle_meta.go

```diff
@@ -67,27 +67,29 @@ func (f FactoryMeta) Process(ctx context.Context, msg *statemachine.DispatcherMe
 		panic(throw.Impossible())
 	}
 
-	mustReject, err := f.AuthService.IsMessageFromVirtualLegitimate(logCtx, payloadObj, payloadMeta.Sender, pr)
-	if err != nil {
-		logger.Warn(throw.W(err, "illegitimate msg", skippedMessage{
-			messageTypeID: payloadTypeID,
-			messageType:   payloadType,
-			incomingPulse: payloadMeta.Pulse,
-			targetPulse:   targetPulse,
-		}))
-
-		return pulse.Unknown, nil, nil
-	}
+	if !pr.RightBoundData().IsExpectedPulse() {
```

Comment:
please, add a comment about what is happening. We will 100% forget about what this check does